src/librustc_yk_sections/emit_tir.rs

```diff
@@ -110,8 +110,72 @@ impl<'a, 'tcx, 'gcx> ConvCx<'a, 'tcx, 'gcx> {
+            TerminatorKind::Goto{target: target_bb}
+            | TerminatorKind::FalseEdges{real_target: target_bb, ..}
+            | TerminatorKind::FalseUnwind{real_target: target_bb, ..} =>
+                ykpack::Terminator::Goto(u32::from(target_bb)),
+            TerminatorKind::SwitchInt{ref discr, ref values, ref targets, ..} => {
+                match self.lower_operand(discr) {
+                    Ok(local) => ykpack::Terminator::SwitchInt{local,
+                        values: values.iter().map(|u| ykpack::SerU128::new(*u)).collect(),
+                        target_bbs: targets.iter().map(|bb| u32::from(*bb)).collect()},
+                    // FIXME dynamic call targets.
+                    Err(_) => ykpack::Terminator::Unimplemented,
+                }
+            },
+            TerminatorKind::Resume => ykpack::Terminator::Resume,
+            TerminatorKind::Abort => ykpack::Terminator::Abort,
+            TerminatorKind::Return => ykpack::Terminator::Return,
+            TerminatorKind::Unreachable => ykpack::Terminator::Unreachable,
+            TerminatorKind::Drop{target: target_bb, unwind: unwind_bb, ..} =>
+                ykpack::Terminator::Drop{
+                    target_bb: u32::from(target_bb),
+                    unwind_bb: unwind_bb.map(|bb| u32::from(bb)),
+                },
+            TerminatorKind::DropAndReplace{target: target_bb, unwind: unwind_bb, ..} =>
+                ykpack::Terminator::DropAndReplace{
+                    target_bb: u32::from(target_bb),
+                    unwind_bb: unwind_bb.map(|bb| u32::from(bb)),
+                },
+            TerminatorKind::Call{ref func, cleanup: cleanup_bb, ref destination, .. } => {
+                let ser_oper = if let Operand::Constant(box Constant {
+                    literal: LazyConst::Evaluated(Const {
+                        ty: &TyS {
+                            sty: TyKind::FnDef(ref target_def_id, ..), ..
+                        }, ..
+                    }), ..
+                }, ..) = func {
+                    // A statically known call target.
+                    ykpack::CallOperand::Fn(self.lower_def_id(target_def_id))
+                } else {
+                    // FIXME -- implement other callables.
+                    ykpack::CallOperand::Unknown
+                };
+
+                let ret_bb = destination.as_ref().map(|(_, bb)| u32::from(*bb));
+                ykpack::Terminator::Call{
+                    operand: ser_oper,
+                    cleanup_bb: cleanup_bb.map(|bb| u32::from(bb)),
+                    ret_bb: ret_bb,
+                }
+            },
+            TerminatorKind::Assert{target: target_bb, cleanup: cleanup_bb, ..} =>
+                ykpack::Terminator::Assert{
+                    target_bb: u32::from(target_bb),
+                    cleanup_bb: cleanup_bb.map(|bb| u32::from(bb)),
+                },
+            TerminatorKind::Yield{resume: resume_bb, drop: drop_bb, ..} =>
```

Comment:
This and some other terminators do not reach codegen: https://github.com/rust-lang/rust/blob/d96c01e77c9a812ed9f101d0e0714e6bd833a0bf/src/librustc_codegen_ssa/mir/block.rs#L879-L883

Reply:
Great stuff! Thanks!